pinot-segment-local/src/main/java/org/apache/pinot/segment/local/function/GroovyFunctionEvaluator.java

```diff
@@ -86,13 +86,13 @@ public List<String> getArguments() {
   public Object evaluate(GenericRow genericRow) {
     for (String argument : _arguments) {
       Object value = genericRow.getValue(argument);
-      if (value == null) {
-        // FIXME: if any param is null a) exit OR b) assume function handles it ?
-        return null;
-      }
       _binding.setVariable(argument, value);
     }
-    return _script.run();
+    try {
+      return _script.run();
```

Comment:
wont errors we were previously catching and throwing during record transformation just get hidden now?

Reply:
@npawar Good call. Yes, it is essentially a decision between fail fast or keep the ingestion going but fill the default value.
Since we are throwing the exception right now, we might want to keep the same behavior. One way to get both worlds is to add a flag to track if any argument is `null`, and return `null` if the flag is `true`, throw exception if the flag is `false`

Reply:
+1 on this approach. I think it's useful to have the ability to throw exception when transformation is problematic.

Reply:
thanks, modifying!